snekbox/nsjail.py

```diff
@@ -163,9 +200,27 @@ def python3(
                 "--",
                 self.config.exec_bin.path,
                 *self.config.exec_bin.arg,
-                *py_args,
-                code,
-            )
+            ]
+
+            # Filter out empty strings (causes issues with python cli)
+            args.extend(s for s in py_args if s)
+
+            c_arg = "c" in "".join(py_args)
+
+            # Override for `timeit`
+            if "timeit" in py_args:
+                use_file = False
+
+            match (use_file, c_arg):
+                case (True, _) | (None, False):
+                    args.append("main.py")
+                    # Write the code to a file
+                    with fs.allow_write():
+                        code_path = fs.home / "main.py"
+                        code_path.write_text(code)
+                        log.info(f"Created code file at [{code_path!r}].")
+                case _:
+                    args.append(code)
 
```

Comment:
This feels pretty fragile, changing behaviour just based on whether there is a letter `c` in the arguments and having to hardcode support in for timeit isn't great.

Maybe we should make the `input` API parameter optional, and then only add `main.py` if that was passed, otherwise just use the args given (which could contain code). This would be pretty simple behaviour and also flexible. Not sure how we deal with breaking changes for snekbox though.

Reply:
Yeah I agree this isn't ideal, I was just trying to make this work without breaking changes mostly.

The timeit situation mainly stems from the fact that it requires string input
```py
{
	"input": "[i for i in range(100)]",
	"args": ["-m", "timeit"]
}
->
python3 -m timeit "i for i in range(100)"
```
whereas something like `unittest` would requires a module/file
```py
python3 -m unittest main.py
```

So, not sure how we can really decide on snekbox side, maybe this should be something in the input request?

Reply:
Optionally we can just fix `input` to always write a file, and if the user wanted `-c` or `timeit` strings, require that to be in `args` instead?

Reply:
Implemented new api format in [ba6d0a8](https://github.com/python-discord/snekbox/pull/159/commits/ba6d0a8a10af687393134fc1e9662100ce67df52) which avoids parsing / guessing on snekbox's side in favor of the client providing all arguments and files. 

```json
"args": ["main.py"],
"files": [
    {
        "name": "main.py",
        "content": "print(1)"
    }
]
```